src/Dictionaries/MongoDBDictionarySource.cpp

```diff
@@ -1,196 +1,150 @@
-    if (!uri.empty())
-    {
-        // Connect with URI.
-        connection->connect(uri, socket_factory);
-
-        Poco::URI poco_uri(connection->uri());
-
-        // Parse database from URI. This is required for correctness -- the
-        // cursor is created using database name and collection name, so we have
-        // to specify them properly.
-        db = poco_uri.getPath();
-        // getPath() may return a leading slash, remove it.
-        if (!db.empty() && db[0] == '/')
-        {
-            db.erase(0, 1);
-        }
-
-        // Parse some other parts from URI, for logging and display purposes.
-        host = poco_uri.getHost();
-        port = poco_uri.getPort();
-        user = poco_uri.getUserInfo();
-        if (size_t separator = user.find(':'); separator != std::string::npos)
-        {
-            user.resize(separator);
-        }
-    }
-    else
-    {
-        // Connect with host/port/user/etc through constructing the uri
-        std::string uri_constructed("mongodb://" + host + ":" + std::to_string(port) + "/" + db + (options.empty() ? "" : "?" + options));
-        connection->connect(uri_constructed, socket_factory);
-
-        if (!user.empty())
-        {
-            Poco::MongoDB::Database poco_db(db);
-            if (!poco_db.authenticate(*connection, user, password, method.empty() ? Poco::MongoDB::Database::AUTH_SCRAM_SHA1 : method))
-                throw Exception(ErrorCodes::MONGODB_CANNOT_AUTHENTICATE, "Cannot authenticate in MongoDB, incorrect user or password");
-        }
-    }
 }
 
-
 MongoDBDictionarySource::MongoDBDictionarySource(const MongoDBDictionarySource & other)
-    : MongoDBDictionarySource{
-        other.dict_struct, other.uri, other.host, other.port, other.user, other.password, other.method, other.db,
-        other.collection, other.options, other.sample_block
-    }
+    : MongoDBDictionarySource{other.dict_struct, other.configuration, other.sample_block}
 {
 }
 
 MongoDBDictionarySource::~MongoDBDictionarySource() = default;
 
 QueryPipeline MongoDBDictionarySource::loadAll()
 {
-    return QueryPipeline(std::make_shared<MongoDBSource>(connection, db, collection, Poco::MongoDB::Document{}, sample_block, max_block_size));
+    return QueryPipeline(std::make_shared<MongoDBSource>(*configuration->uri, configuration->collection, make_document(), mongocxx::options::find(), sample_block, max_block_size));
 }
 
 QueryPipeline MongoDBDictionarySource::loadIds(const std::vector<UInt64> & ids)
 {
     if (!dict_struct.id)
         throw Exception(ErrorCodes::UNSUPPORTED_METHOD, "'id' is required for selective loading");
 
-    Poco::MongoDB::Document query;
-
-    /** NOTE: While building array, Poco::MongoDB requires passing of different unused element names, along with values.
-      * In general, Poco::MongoDB is quite inefficient and bulky.
-      */
+    auto ids_array = array();
+    for (const auto & id : ids)
+        ids_array.append(static_cast<Int64>(id));
```

Comment:
No, because it's defined in the interface:
https://github.com/ClickHouse/ClickHouse/blob/fe2c7efc41dc6248499be5b38b4d10bc9582778f/src/Dictionaries/IDictionarySource.h#L36